Game2048Base/src/main/java/models/grid2048/Grid2048.java

```diff
@@ -0,0 +1,118 @@
+package models.grid2048;
+
+import com.google.common.annotations.VisibleForTesting;
+import javalib.worldimages.OutlineMode;
+import javalib.worldimages.OverlayOffsetImage;
+import javalib.worldimages.Posn;
+import javalib.worldimages.RectangleImage;
+import javalib.worldimages.WorldImage;
+import models.square.EmptySquare;
+import models.square.Square;
+import models.square.Tile;
+import utility.PosnUtility;
+import java.awt.Color;
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public class Grid2048 {
+
+    public Square[][] grid;
+    static WorldImage IMAGE = new RectangleImage(
+            Square.SIDE_LENGTH * 4,
+            Square.SIDE_LENGTH * 4,
+            OutlineMode.OUTLINE,
+            Color.BLACK);
+
+    public Grid2048() {
+        this.grid = initializeStartingGrid();
+    }
+
+    public Grid2048(Square[][] grid) {
+        this.grid = grid;
+    }
+
+    private static Square[][] initializeStartingGrid() {
+        Square[][] grid = new Square[4][4];
+
+        createEmptySquaresOnGrid(grid);
+        addTwoRandomTilesForInitializedGrid(grid);
+
+        return grid;
+    }
+
+    public static void createEmptySquaresOnGrid(Square[][] grid) {
+        for (int idxRow = 0; idxRow < 4; idxRow ++) {
+            for (int idxColumn = 0; idxColumn < 4; idxColumn ++) {
+                grid[idxRow][idxColumn] = new EmptySquare(new Posn(idxRow, idxColumn));
+            }
+        }
+    }
+
+    @VisibleForTesting
+    static void addTwoRandomTilesForInitializedGrid(Square[][] grid) {
+        ArrayList<Posn> tilePosns = PosnUtility.createRandomPosns();
+        Posn t0Posn = new Posn (tilePosns.get(0).x, tilePosns.get(0).y);
+        Posn t1Posn = new Posn (tilePosns.get(1).x, tilePosns.get(1).y);
+        int t0Value = Tile.weightedRandomTileValue();
+        int t1Value = Tile.weightedRandomTileValue();
+        grid[t0Posn.x][t0Posn.y] = new Tile(t0Value, t0Posn);
+        grid[t1Posn.x][t1Posn.y] = new Tile(t1Value, t1Posn);
+    }
+
+    public WorldImage drawGrid() {
+        WorldImage resultGrid = IMAGE;
+        int dxOffset = -2;
```

Comment:
what are these offsets for?

Reply:
I'm sorry I'm just going to keep this "magic" as is the Java library for this is really confusing and I'm over it